src/docfx/build/toc/TableOfContentsLoader.cs

```diff
@@ -37,29 +41,106 @@ internal class TableOfContentsLoader
+            _joinTOCConfigs = config.JoinTOC is null
+                ? new Dictionary<string, JoinTOCConfig>()
+                : config.JoinTOC.Where(x => x.ReferenceToc != null).ToDictionary(x => PathUtility.Normalize(x.ReferenceToc!));
         }
 
         public (TableOfContentsNode node, List<Document> referencedFiles, List<Document> referencedTocs)
-            Load(ErrorBuilder errors, Document file)
+            Load(Document file)
         {
             return _cache.GetOrAdd(file.FilePath, _ =>
             {
                 var referencedFiles = new List<Document>();
                 var referencedTocs = new List<Document>();
-                var node = LoadTocFile(file, file, referencedFiles, referencedTocs, errors);
+                var node = LoadTocFile(file, file, referencedFiles, referencedTocs, _errors);
 
+                if (_joinTOCConfigs.TryGetValue(file.FilePath.Path, out var joinTOCConfig) && joinTOCConfig != null)
+                {
+                    if (joinTOCConfig.TopLevelToc != null)
+                    {
+                        node = JoinToc(node, joinTOCConfig.TopLevelToc);
+                    }
+                }
                 return (node, referencedFiles, referencedTocs);
             });
         }
 
+        private TableOfContentsNode JoinToc(TableOfContentsNode referenceToc, string topLevelTocFilePath)
+        {
+            var (topLevelToc, _, _) = Load(_documentProvider.GetDocument(FilePath.Content(new PathString(topLevelTocFilePath))));
+
+            var itemsToMatch = new Dictionary<string, SourceInfo<TableOfContentsNode>>();
+            var duplicatedName = new HashSet<string>();
+            foreach (var child in referenceToc.Items)
+            {
+                var key = child.Value.Name;
+                if (string.IsNullOrEmpty(key))
+                {
+                    _errors.Add(Errors.JsonSchema.MissingAttribute(key, "name"));
+                }
+                else if (!duplicatedName.Add(key!))
+                {
+                    _errors.Add(Errors.TableOfContents.DuplicateJoinTocName(key!));
+                    itemsToMatch.Remove(key!);
+                }
+                else
+                {
+                    itemsToMatch.Add(key!, child);
+                }
+            }
+
+            TraverseAndMerge(topLevelToc, itemsToMatch);
+            return topLevelToc;
+        }
+
+        private void TraverseAndMerge(TableOfContentsNode node, Dictionary<string, SourceInfo<TableOfContentsNode>> itemsToMatch)
+        {
+            if (node.ExtensionData.TryGetValue("children", out var token) && token is JArray array)
+            {
+                node.ExtensionData.Remove("children");
+                foreach (JValue v in array)
+                {
+                    if (v.Value is string str)
+                    {
+                        var keysMatched = new HashSet<string>();
+                        var pattern = str[0..^1];
```

Comment:
What is the definition of the pattern, regex or wildcard or glob? Trimming the last char doesn't sound right.